datafusion/core/src/physical_planner.rs

```diff
@@ -3106,7 +3106,7 @@ mod tests {
 
         assert_contains!(
             &e,
-            r#"Error during planning: Can not find compatible types to compare Boolean with [Struct("foo": Boolean), Utf8]"#
+            r#"Error during planning: Can not find compatible types to compare Boolean with [Struct("foo": non-null Boolean), Utf8]"#
```

Comment:
these are due the changes from https://github.com/apache/arrow-rs/issues/8648 to clean up datatype display. It is a nice improvement in my mind